rest_framework/schemas/inspectors.py

```diff
@@ -285,7 +286,10 @@ def get_serializer_fields(self, path, method):
         if not hasattr(view, 'get_serializer'):
             return []
 
-        serializer = view.get_serializer()
+        try:
+            serializer = view.get_serializer()
+        except exceptions.APIException:
+            serializer = None
```

Comment:
I think we need a warning here. Something like 

    """
    {}.get_serializer() raised an exception during schema generation. 
    It will be excluded for {} {}.
    """.format(view.cls.__name__, method, path)

(The formatting there is more for GitHub than the code.)

This way we'll have some kind of console feedback so users can address any issues.

Reply:
Good idea, added a warning.